### PR TITLE
 Fix flaky time travel spec

### PR DESCRIPTION
**Pull request by jkempster34**
## Changes in this PR

This spec is flaky in CI.

I'm not sure why, but sometimes when time travelling, the user gets signed
out, and the next assertion fails.

This re-authenticates the user after time travelling. It also freezes the time
in the correct before block, as previously it was accidentally part of a different
context.

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [ ] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

File: spec/features/users_can_edit_a_budget_spec.rb [hunk 1]
```diff
@@ -92,29 +94,30 @@
...
+      travel_to(Date.tomorrow)
+      authenticate!(user: user)
 
-        expect(page).to have_content(t("action.budget.update.success"))
-        within("##{budget.id}") do
-          expect(page).to have_content("£20.00")
-          expect(page).to have_link("1 revision", href: activity_budget_revisions_path(budget.parent_activity_id, budget))
-          click_on t("default.link.edit")
-        end
+      click_on t("default.link.edit")
```
Comment (yndajas) [inline spec/features/users_can_edit_a_budget_spec.rb:102]: Does this need to be in `within("##{budget.id}") do` like before, or is there only one budget?

File: spec/features/users_can_edit_a_budget_spec.rb [hunk 2]
```diff
@@ -127,21 +130,21 @@
       within(original_row) do
         expect(page).to have_content("£10.00")
         expect(page.all("td", exact_text: "").count).to be 2
-        expect(page).to have_content(I18n.l(Date.today))
+        expect(page).to have_content(I18n.l(Date.yesterday))
```
Comment (yndajas) [inline spec/features/users_can_edit_a_budget_spec.rb:133]: This makes sense to me now, but I don't understand (without putting much thought into it) why it was `Date.today` before, given the time travelling 🤔

**Review by yndajas — APPROVED** [review]
One question and one confusion, but looks like this is working in any case

File: spec/features/users_can_edit_a_budget_spec.rb (hunk 2 above)
Comment (jkempster34) [reply spec/features/users_can_edit_a_budget_spec.rb:136]: Before, the time travelling was in a block, so it was returning to the current time outside of the block 🤯

**Pull request merged**